Updated goccy/go-yaml to v.14.1

XuechunHou opened the pull request:
## Description
Updated `goccy/go-yaml` to `v.14.1` by executing the command `go get -u github.com/goccy/go-yaml@v1.14.1`

## Related issue
b/376658816

## How has this been tested?
<!--- Please describe how you tested the changes besides the automatically triggered unit tests when applicable. -->
<!--- Must include sample output logs or metrics and/or screenshots of key results when applicable. -->

## Checklist:
- Unit tests
  - [] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [X] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [X] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [X] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

<!--- To edit this template, go to https://github.com/GoogleCloudPlatform/ops-agent/edit/master/.github/PULL_REQUEST_TEMPLATE.md -->

braydonk reviewed (commented):
Thanks for doing this. Looks like there is a breaking change in the pretty printing that this PR caught. I'm going to open an issue upstream about this. Our update can be on hold until then.

braydonk commented:
Opened https://github.com/goccy/go-yaml/issues/534

braydonk commented:
@XuechunHou The upstream issue has been opened. Can you update the package to `v1.14.1` and update the goldens?

franciscovalentecastro commented on integration_test/metadata/integration_metadata.go:
```diff
@@ -155,7 +155,7 @@ func UnmarshalAndValidate(fullPath string, data []byte, i interface{}) error {
 	v := NewIntegrationMetadataValidator()
 	// Note: Unmarshaler does not protect when only the key being declared.
 	// https://github.com/goccy/go-yaml/issues/313
```
Please update this comment if this issue (# 313) has been fixed with the current update.

braydonk replied:
Won't be fixed until 1.14.3, which is available if we want to update to that.

franciscovalentecastro reviewed (approved):
LGTM modulo the integration test pass and addressing comments.

franciscovalentecastro commented on transformation_test/transformation_test.go:
```diff
@@ -569,7 +569,7 @@ func sanitizeWriteLogEntriesRequest(t *testing.T, r *logpb.WriteLogEntriesReques
 	// We need to remove non-deterministic information from stacktraces so the goldens don't keep changing.
 	// Remove $GOPATH
-	result := regexp.MustCompile(`(?m)^\t(.*?)pkg/mod/`).ReplaceAllString(input, "\t")
+	result := regexp.MustCompile(`(?m)^\t(.*?)pkg/mod/`).ReplaceAllString(input, "  ")
```
I'm curious, why did it has to change from tabs (`\t`) to two spaces ?

braydonk replied:
Using `\t` (tab) where we did is illegal according to the YAML spec. Tabs can't be used as indentation, and putting the `\t` character where we did ended up mixing space and tab indentation. The old version of the YAML library parsed this incorrectly, allowing it to be possible when it should be illegal. This is fixed in this newer version of the library. I decided on two spaces to maintain the original intent of the replacement while still making it legal YAML.

XuechunHou commented:
> LGTM modulo the integration test pass and addressing comments.

Ops Agent and Third party integration tests are failing, though they are already failing at Master head.  Also, these failed tests on github are passing on Louhi.  

These tests mostly failed because "context deadline exceeded"
